### PR TITLE
add SeqIR node

### PR DESCRIPTION
**Pull request by avibryant**
This adds the `SeqIR` node which has two children, `first` and `second`, which it runs in sequence and then returns the value of `second`. This seems like a strange thing to have in a pure expression language - what's the point of running `first` if we're throwing away the result? The goal here is effectively to act as a "cache warmer" - in particular, for (future) large n-ary nodes like `Lookup` and `Sum`, it's useful to schedule the computation of each of their children in advance so that the code they ultimately generate can be as compact as possible (just referencing pre-computed values), which means we can pack as large a node as possible into a single JIT-friendly method.

Because of this, `first` and `second` are constrained to be `VarDef`, since there's no point in pre-computing a `Ref`.

### CODE REVIEW

**Comment by avibryant** [comment]
Added a WIP because as I'm trying to use this I'm becoming unsure about the types.

File: rainier-core/src/main/scala/com/stripe/rainier/ir/Tracer.scala [hunk 1]
```diff
@@ -97,6 +97,7 @@ object Tracer {
             s"if($t == 0.0) $z else $nz"
           else
             s"(if($t == 0.0) $z else $nz)"
+        case s: SeqIR => ???
```
Comment (mio-stripe) [inline rainier-core/src/main/scala/com/stripe/rainier/ir/Tracer.scala]: This goes away with the new tracer, right?

Comment (avibryant) [reply rainier-core/src/main/scala/com/stripe/rainier/ir/Tracer.scala]: correct, which is why I left it this way :)

Comment (avibryant) [reply rainier-core/src/main/scala/com/stripe/rainier/ir/Tracer.scala]: I'll do the rebase.

**Comment by avibryant** [comment]
Ah actually merging the Lookup branch in fixed the need for a rebase :)

**Pull request merged**